apps/extension/src/hooks/full-sync-height.ts

```diff
@@ -55,3 +55,14 @@ export const useSyncProgress = () => {
 
   return { latestBlockHeight, fullSyncHeight, error };
 };
+
+export const fetchBlockHeight = async (grpcEndpoint: string) => {
+  const tendermintClient = createPromiseClient(
+    TendermintProxyService,
+    createGrpcWebTransport({ baseUrl: grpcEndpoint }),
+  );
+  const blockHeight = (await tendermintClient.getStatus({}).catch(() => undefined))?.syncInfo
+    ?.latestBlockHeight;
+
+  return blockHeight ? Number(blockHeight) : undefined;
+};
```

Comment:
issue: this seems to be redundant to the function nested in the hook above. However, looks like this one tries to return an `undefined` in a failure case (don't think that's necessary given the consumer uses a try/catch). Can we extract the original and use it for both places?

Reply:
good call, I refactored `useSyncProgress` to call `fetchBlockHeight` to reduce code duplication.

Reply:
modified to use react query pattern